[#18] Adds formatting for services and an additional example

hydroid7 opened the pull request:
## Notes for Reviewer
Here is the pull request with fixed branch name and better formatting. 

## Pre-Review Checklist for the PR Author

1. [ ] Add sensible notes for the reviewer
1. [x] PR title is short, expressive and meaningful
1. [x] Relevant issues are linked
1. [x] Every source code file has a copyright header with `SPDX-License-Identifier: Apache-2.0 OR MIT`
1. [x] Branch follows the naming format (`iox2-123-introduce-posix-ipc-example`)
1. [ ] Commits messages are according to this [guideline][commit-guidelines]
    - [ ] Commit messages have the issue ID (`[#123] Add posix ipc example`)
    - [ ] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [ ] Tests follow the [best practice for testing][testing]
1. [ ] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [ ] Assign PR to reviewer
1. [ ] All checks have passed (except `task-list-completed`)

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx2/blob/main/doc/release-notes/iceoryx2-unreleased.md

## Checklist for the PR Reviewer

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Unit tests have been written for new behavior
- [ ] Public API is documented
- [ ] PR title describes the changes

## Post-review Checklist for the PR Author

1. [ ] All open points are addressed and tracked via issues

## References

Use either 'Closes #123' or 'Relates to #123' to reference the corresponding issue.

Closes #29

elfenpiff commented on examples/examples/pretty_print.rs:
```diff
@@ -0,0 +1,11 @@
+use iceoryx2::service::{service_name::ServiceName, zero_copy, Service};
```
I think this is a feature that does not require a separate example.

hydroid7 replied:
Sounds reasonable.

elfenpiff reviewed (changes requested):
Thank you very much for the contribution :smile: 

Could you please adjust the commit messages and add `[#29] ` as prefix to every commit message.
You can do this by calling `git rebase -i HEAD~1`, set the commit in editor from `pick` to `edit`, adjust the comment with `git commit --amend` and then call `git rebase --continue`.

This you can then push with `git push -f`.

When this is done and the comments are addressed we can merge this thing.

And welcome to the team, you are our first external contributor!

elfenpiff commented on iceoryx2/src/service/port_factory/event.rs:
```diff
@@ -127,3 +130,53 @@ impl<'config, Service: service::Details<'config>> PortFactory<'config, Service>
     }
 }
+
+impl<'config> fmt::Display for PortFactory<'config, zero_copy::Service<'config>> {
```
I think it would be better to introduce the second generic parameter here: `Service: service::Details<'config>` and then replace the line `f.debug_struct("zero_copy::Service")` (136) with

```rust
f.debug_struct(format!("{}", core::any::type_name::<Service>()))
```

With this we reduce the code duplication and when we introduce another service variant we do not have to add here a third version (which may is even forgotten).

@hydroid7 What do you think?

hydroid7 replied:
Oh, `core::any::type_name` is a game changer here! :tada:

elfenpiff commented on iceoryx2/src/service/port_factory/event.rs:
```diff
@@ -127,3 +130,53 @@ impl<'config, Service: service::Details<'config>> PortFactory<'config, Service>
+    }
+}
+
+impl<'config> fmt::Display for PortFactory<'config, process_local::Service<'config>> {
```
@hydroid7 If you follow my suggestion from above we can remove this here.

elfenpiff reviewed (commented):
Looks good. Could you please also add the `Display` implementation to the port factory of the publish subscribe messaging pattern in `src/service/port_factory/publish_subscribe.rs`.

elfenpiff commented:
@hydroid7 could you please also add this into the `doc/release-notes/iceoryx2-unreleased.md` release documentation under the `features` section.

The entry would look somehow like this:
```
* Add `Display` implementation for `PortFactory` [#29](https://github.com/eclipse-iceoryx/iceoryx2/issues/29)
```

elBoberido commented:
I did not review the patch but please don't forget why this is done. We want to replace the custom printing code in these to locations
- https://github.com/eclipse-iceoryx/iceoryx2/blob/main/iceoryx2/src/service/port_factory/publish_subscribe.rs#L24-L34
- https://github.com/eclipse-iceoryx/iceoryx2/blob/main/iceoryx2/src/service/port_factory/event.rs#L24-L29

Pleas change those examples to just use `println!("{}", pubsub);` and `println!("{}", event);`

Oh, and the PR and commit messages use the wrong issue number. It's actually #18. #29 was the previous PR.

hydroid7 commented:
Thank you all for the constructive comments, I rewrote the commit messages, added the feature to the releasenotes, changed the documentation in order to reflect the new `Display` implementation and added the same for `pubsub` services.

elBoberido commented on doc/release-notes/iceoryx2-unreleased.md:
```diff
@@ -6,7 +6,7 @@
 ### Features
 
- * Example [#1](https://github.com/larry-robotics/iceoryx2/issues/1)
+ * Add `Display` implementation for `PortFactory` [#29](https://github.com/eclipse-iceoryx/iceoryx2/issues/18)
```
```suggestion
 * Add `Display` implementation for `PortFactory` [#18](https://github.com/eclipse-iceoryx/iceoryx2/issues/18)
```

elBoberido reviewed (commented):
Only had a look at the release notes. I'll leave the actual code to @elfenpiff

elfenpiff reviewed (commented):
Looks good, as soon as the issue number in the release notes is corrected we can merge this.

hydroid7 commented:
Sorry that these took that long, but here they are.

Background for my contributions is that I'm writing a DDS middleware / message broker as my MSc Thesis.
Therefore I was first looking at iceoryx, but it was rather hard to use for someone without extensive knowledge in the project.

elBoberido reviewed (commented):
Please add `[#18]` here, similar to the other PR.

elBoberido commented:
@hydroid7 you need to fix the clippy warnings to make the CI happy

elBoberido commented:
@hydroid7 do you plan to finish this PR in the near future?

elfenpiff commented:
I think the PR is abandoned for now. @hydroid7 if you would like to continue working on this please feel free to reopen the PR again.
